pages/calendar.js

```diff
@@ -1,28 +1,54 @@
-"use client";
-import React from "react";
 import "react-calendar/dist/Calendar.css";
-import activitiesData from "@/lib/db";
 import dynamic from "next/dynamic";
 import styled from "styled-components";
 import useAuthentication from "./api/auth/useAuthentication";
+import Navigation from "@/components/Navigation";
+import { useState } from "react";
+import Link from "next/link";
+import ActivityCard from "@/components/ActivityCard";
+import { theme } from "@/styles";
+import Logo from "@/Icons/Logo";
+import useSWR from "swr";
 
 const Calendar = dynamic(() => import("react-calendar"), { ssr: false });
 
 const CalendarPage = () => {
+  const { data: activities, isLoading } = useSWR("/api/activities");
   const { authenticated, loading } = useAuthentication();
-  const activitiesDates = activitiesData.map((activity) => activity.date);
-  const isActivityDate = (date) => {
-    const dateString = date.toISOString().split("T")[0];
-    return activitiesDates.includes(dateString);
+  const [selectedDateActivities, setSelectedDateActivities] = useState([]);
+
+  if (isLoading) return <div>loading...</div>;
+  if (!activities) return <div>failed to load</div>;
+
+  const getActivitiesForDate = (date) => {
+    const selectedYear = date.getFullYear();
+    const selectedMonth = date.getMonth();
+    const selectedDay = date.getDate();
+
+    return activities.filter((activity) => {
+      const activityDate = new Date(activity.date);
+      return (
+        activityDate.getFullYear() === selectedYear &&
+        activityDate.getMonth() === selectedMonth &&
+        activityDate.getDate() === selectedDay
+      );
```

Comment:
You could simplify this by comparing the "dateStrings" of the two dates: 
```js
date.toDateString() === activityDate.toDateString()
```

Here is a quick reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toDateString